docs/engineering-defaults.md

```diff
@@ -1,7 +1,151 @@
 # Engineering Defaults
 
+Much of OpenO11y's guidance is based on or enhanced by these best practices for
+engineering teams.
+
 ## Pair Programming
```

Comment:
A bit of structure for these will help
- Definition
- How to do X (default)
- How to measure (connecting it back to metrics)